core/frontend/src/WebViewerApp.ts

```diff
@@ -55,7 +54,11 @@ export class WebViewerApp {
         await BrowserAuthorizationCallbackHandler.handleSigninCallback(redirectUrl);
         return;
       }
+    }
 
+    await IModelApp.startup(opts.iModelApp);

```

Comment:
Not sure I understand the need for this still. Is this just to speed up the redirect handling so iModelApp doesn't need to be instantiated before we just redirect?

Reply:
It's checking for a _callback_, not a _redirect_ before IModelApp.startup.
This is significant, not on a redirection, but in silent and popup signins primarily, since we want to succeed or fail out of those as quickly as possible, not startup IModelApp every time.

I actually think we might be able to remove that **return** after handling the callback, since that redirect callbacks don't necessarily kill the window. I can look into that further too.

Reply:
Yea, sounds good I meant the redirect callback. That works with me.